### ISSUE TITLE
Custom configuration in Config directory itself - animdl_config.yml

### ISSUE THREAD

**Issue by Anu-bhav** [justfoolingaround/animdl#80]
Hi @justfoolingaround 

I had an issue with the way animdl loads the custom config `animdl_config.yml`. 

At the moment, it loads from either from `os.getenv('ANIMDL_CONFIG')` or from the `current directory` that we execute animdl `./animdl_config.yml`. 

I did not want to hassle setting ENV variable in windows. 
And as you can see the flaw that if we execute animdl from any other directory, the custom configs do not take effect. 

I misunderstood the docs at first and did not know why my config did not take effect as I created them in both the root folder and in the config folder as well.
I added pprint to know if my config is merged or not.
![image](https://user-images.githubusercontent.com/41940586/143506344-94a0c35a-8826-409c-a945-291a43d53636.png)

Then I went to examine the code and found the issue.

My proposed solution is to search for the `animdl_config.yml` in the config directory itself first.
![image](https://user-images.githubusercontent.com/41940586/143506576-89d73338-a9df-485c-8ec2-92d688610233.png)

I can now run animdl from anywhere and you can now see, it did merge the config.
The code that does the magic:

```CONFIGURATION_FILE_PATH = Path(os.getenv("ANIMDL_CONFIG") or os.path.dirname(os.path.realpath(__file__)) + "/animdl_config.yml" or "./animdl_config.yml")```

My custom config:
```
default_provider: "animixplay"
preferred_quality: 720
default_player: "vlc"
players:
  vlc:
    executable: "vlc"
    opts: 
      - --network-caching=5000
      - --play-and-exit
      - --fullscreen
```

If you want I can make a pull request for you to accept the solution. I will exclude my configuration. 
However for VLC users, it is quite a good configuration as it starts in fullscreen. 
When finished, it autoexits and then the next video starts automatically.
You can provide my config in the docs as a sample for VLC users.

**Comment by justfoolingaround** [comment]
I've added a solution to this problem. By default, the project used to look for `animdl_config.yml` <b>on the root directory</b>.

The changes I've made have made it so that, the project will go around 4 potential candidates to load and overwrite the default configurations.

These candidates (in their priority order) are as follows:

- Path stored in the environment variable `ANIMDL_CONFIG` (made so that animdl can be used in remote servers with easier configuration loading.)
- `animdl_config.yml` **in working directory**
- `animdl_config.yml` **in animdl/core/config** (From your suggestion)
- `%USERPROFILE%/.animdl/config.yml` (Windows) or `./.animdl/config.yml` (Linux)

Configurations will not be loaded from all of these candidates, like I mentioned earlier, they will be loaded from a configuration file that is available first.

There is a problem in your implementation however.

```py
CONFIGURATION_FILE_PATH = Path(os.getenv("ANIMDL_CONFIG") or os.path.dirname(os.path.realpath(__file__)) + "/animdl_config.yml" or "./animdl_config.yml")```

This line will **never** load from `./animdl_config.yml`. Python `OR` does not determine whether the path is available or not, it goes by the boolean nature of the object. `os.path.dirname(os.path.realpath(__file__)) + "/animdl_config.yml"` will always give `True` and hence, will always get selected.

Thank you for your suggestion, some of it has made into the project!
